alternator/executor.cc

```diff
@@ -1395,7 +1395,8 @@ static lw_shared_ptr<query::read_command> previous_item_read_command(service::st
     auto regular_columns = boost::copy_range<query::column_id_vector>(
             schema->regular_columns() | boost::adaptors::transformed([] (const column_definition& cdef) { return cdef.id; }));
     auto partition_slice = query::partition_slice(std::move(bounds), {}, std::move(regular_columns), selection->get_query_options());
-    return ::make_lw_shared<query::read_command>(schema->id(), schema->version(), partition_slice, proxy.get_max_result_size(partition_slice));
+    return ::make_lw_shared<query::read_command>(schema->id(), schema->version(), partition_slice, proxy.get_max_result_size(partition_slice),
+            query::tombstone_limit(proxy.get_tombstone_limit()));
```

Comment:
nitpick: is there any reason why `proxy.get_tombstone_limit()` returns a bare integer instead of this `tombstone_limit` type? It would be more natural to just return the correct type right away

Reply:
I guess it made life harder for code using the IDL constructor. Maybe the time came for me to move all code to the new constructor.